### PR TITLE
[Feature] Adds mixin blueprint for module unification

### PR DESCRIPTION
**Pull request by beegan**
Part of https://github.com/ember-cli/ember-cli/issues/7530

### CODE REVIEW

**Comment by GavinJoyce** [comment]
Nice one @jackbeegan, this is looking great.

**Comment by GavinJoyce** [comment]
@jackbeegan and I had a chat about his idea to increase test coverage to cover the following currently untested scenarios:

 * [ ] in-repo addons
 * [ ] dummy apps

It would be valuable to add this coverage in this PR, we can then use this as an example to increase coverage for all other blueprints

**Comment by GavinJoyce** [comment]
@jackbeegan I added some `ember g service` tests for in-repo addons in classic apps: https://github.com/emberjs/ember.js/pull/16553. We can't yet write similar tests for MU apps as we haven't added MU support to `ember g in-repo-addon` yet. I'll get that done sone.

I also noticed that ember don't seem to support `ember g service --dummy`. I think we can safely live without dummy app generation tests for now, I'll follow up at the next st-module-unification weekly meeting and verify if it would be valuable to support this or not. Either way, we can ignore them for now

File: blueprints/mixin/index.js [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+        __root__(options) {
+          if (options.pod) {
+            throw "Pods aren't supported within a module unification app";
+          }
+
+          return 'src';
+        },
+        __testType__() {
+          return 'mixins';
+        },
```
Comment (GavinJoyce) [inline blueprints/mixin/index.js]: ⚡️ ?

File: node-tests/blueprints/mixin-test.js [hunk 2]
```diff
@@ -197,4 +290,53 @@ describe('Blueprint: mixin', function() {
       });
     });
   });
+
+  describe('in in-repo-addon - module unification', function() {
```
Comment (GavinJoyce) [inline node-tests/blueprints/mixin-test.js]: I haven't yet landed https://github.com/ember-cli/ember-cli/pull/7783 so you can remove this test for now, it's not currently possible to test the desired behaviour of MU `in-repo-addon` blueprint output. I'll be sure to follow up and add these tests later

**Comment by GavinJoyce** [comment]
Nice one @jackbeegan, this looks great. As a last step, perhaps you could squash and rebase?

**Comment by GavinJoyce** [comment]
@jackbeegan as a last step, can you prefix the commit message with `[Feature]`?

**Review by GavinJoyce — APPROVED** [review]
CI Failures seems to be flakiness

👍 nice one @jackbeegan

**Comment by GavinJoyce** [comment]
@rwjblue this looks good to me and is ready for review. It seems that the test failure is just flaky CI

**Review by rwjblue — APPROVED** [review]

**Comment by rwjblue** [comment]
Thank you!!

**Pull request merged**